publisher.py

```diff
@@ -340,25 +352,29 @@ def fetch_image(self, image: Image):
 
         Each `Registry` has its own client so we just access it directly.
         """
-        self.log.warning("Fetching image %s", image)
+        self.log.info("Fetching image %s", image)
         for line in image.repository.registry.client.pull(str(image.repository), image.tag, stream=True, decode=True):
-            self.log.warning(line)
+            self.log.debug(line)
 
     def tag_image(self, image: Image, target_repository: Repository, tag: str):
-        self.log.warning("Tagging %s for %s:%s", image, target_repository, tag)
+        self.log.info("Tagging %s for %s:%s", image, target_repository, tag)
         image.repository.registry.client.tag(str(image), str(target_repository), tag)
 
     def publish_image(self, target: Registry, repository: str, tag: str):
-        self.log.warning("Pushing %s:%s to %s", repository, tag, target)
+        self.log.info("Pushing %s:%s to %s", repository, tag, target)
         if self.dry_run:
             return
         for line in target.client.push(repository, tag, stream=True, decode=True):
-            self.log.warning(line)
+            if 'error' in line.keys():
+               response = int(re.search('\d{3}', line['error'])[0])
```

Comment:
What about connection timeout or dns resolution error? When there is no error number the list index access will throw.

Reply:
Ok, I give up on parsing that (too many different issues and API doesn't care) so I'll just log the errorMessage.